dartagnan/src/main/java/com/dat3m/dartagnan/program/event/rmw/cond/RMWReadCond.java

```diff
@@ -28,12 +31,14 @@
     }
 
     @Override
-    public void initialise(VerificationTask task, Context ctx) {
+    public void initialise(VerificationTask task, SolverContext ctx) {
         super.initialise(task, ctx);
-        z3Cond = ctx.mkEq(memValueExpr, cmp.toZ3Int(this, ctx));
+        z3Cond = ctx.getFormulaManager().getIntegerFormulaManager().equal(
+        		(IntegerFormula)memValueExpr, 
+        		(IntegerFormula)cmp.toIntFormula(this, ctx));
```

Comment:
Is cmp.toIntFormula guaranteed to be an IntegerFormula (is it never a BitVectorFormula?)